### ISSUE TITLE
Unable to generate the jar for mysql 5.7.16 windows

### ISSUE THREAD

**Issue by shrutiujjwal** [mike10004/mysql-dists#1]
Unable to generate the jar for mysql 5.7.16 windows, gives error:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.5.3:single (distro-assembly) on project mysql-dist: Failed to create assembly: Error creating as
ly archive windows-amd64: You must set at least one file. -> [Help 1]

Can you generate the jar & give me pls -or- tell me how to fix.

**Comment by mike10004** [comment]
To clarify, you need MySQL version 5.7.16 instead of 5.7.14?

**Comment by shrutiujjwal** [comment]
Yes Correct. I am having issues using 5.7.14 so thought of trying 5.7.16 -- but wasnt able to create that zip. 

On WINDOWS --> When I use MySQL 5.7.14 that you have put in Git alongwith JCABI I get following error: 

[INFO] port 51379 is available after 4s of waiting
[INFO] $ C:\Perforce_Workspace\ws3\InMemDb\target\mysql-dist\bin\mysqladmin.exe --no-defaults --port=51379 --user=root --socket=C:\Perforce_Workspace\ws3\InMemDb\target\mysql-data\mysql.sock --host=127.0.0.1 password root
[WARNING] >> mysqladmin: connect to server at '127.0.0.1' failed
[WARNING] >> error: 'Host 'localhost' is not allowed to connect to this MySQL server'
[INFO] $ C:\Perforce_Workspace\ws3\InMemDb\target\mysql-dist\bin\mysql.exe --port=51379 --user=root --password=root --socket=C:\Perforce_Workspace\ws3\InMemDb\target\mysql-data\mysql.sock
[WARNING] >> mysql: [Warning] Using a password on the command line interface can be insecure.
[WARNING] >> ERROR 1130 (HY000): Host 'localhost' is not allowed to connect to this MySQL server
[INFO] #start('Config(tcpport=51379, dbuser=root, dbpassword=root, name=root, dbopts=[])', C:\Perforce_Workspace\ws3\InMemDb\target\mysql-dist, C:\Perforce_Workspace\ws3\InMemDb\target\mysql-data): in 12s

**Comment by mike10004** [comment]
If version 5.7.14 satisfies your needs, then you should avoid creating a new artifact for a different version. Try what I suggest in the thread at https://github.com/mike10004/mysql-dists/issues/2.

**Comment by shrutiujjwal** [comment]
Hi Mike, 
Thanks for suggesting the fix. When I updated the POM and tried:
* The Zip was downloaded from: com.github.mike10004 --> but because it did not have scripts folder under it with: scripts\mysql_install_db.pl -- so it threw an error for this one.
* Then I took the scripts folder from the JCABI MYSQL (5.6.14) and tried, it gave the following error: 
[INFO] >> 2016-11-21T18:33:42.915159Z 0 [Note] C:\Perforce_Workspace\ws3\InMemDb\target\mysql-dist\bin\mysqld.exe: ready for connections.
[INFO] >> Version: '5.7.14-log'  socket: ''  port: 55673  MySQL Community Server (GPL)
[INFO] port 55673 is available after 4s of waiting
[INFO] $ C:\Perforce_Workspace\ws3\InMemDb\target\mysql-dist\bin\mysqladmin.exe --no-defaults --port=55673 --user=root --socket=C:\Perforce_Workspace\ws3\InMemDb\target\mysql-data\mysql.sock --host=127.0.0.1 password root
[WARNING] >> mysqladmin: connect to server at '127.0.0.1' failed
[WARNING] >> error: 'Host 'localhost' is not allowed to connect to this MySQL server'
[INFO] $ C:\Perforce_Workspace\ws3\InMemDb\target\mysql-dist\bin\mysql.exe --port=55673 --user=root --password=root --socket=C:\Perforce_Workspace\ws3\InMemDb\target\mysql-data\mysql.sock
[WARNING] >> mysql: [Warning] Using a password on the command line interface can be insecure.
[WARNING] >> ERROR 1130 (HY000): Host 'localhost' is not allowed to connect to this MySQL server
[INFO] #start('Config(tcpport=55673, dbuser=root, dbpassword=root, name=root, dbopts=[])', C:\\Perforce_Workspace\\ws3\\InMemDb\\target\\mysql-dist, C:\\Perforce_Workspace\\ws3\\InMemDb\\target\\mysql-data): in 13s

Pls refer the detailed logs: 
[ErrorLog.txt](https://github.com/mike10004/mysql-dists/files/604481/ErrorLog.txt)

**Comment by shrutiujjwal** [comment]
We did some more investigation of MySQL 5.7.14/16 on Windows Box and below are our findings:

1). In the etc\hosts file I have the entry:

3.204.166.180	[the hostname of my machine]
//127.0.0.1       localhost -- commented
//::1             localhost -- commented


2). Then when I run the 'maven install' command at the target --> --- jcabi-mysql-maven-plugin:1.0r2:start (mysql-test) @ InMemDb ---

At the line it throws following error::
[INFO] >> Version: '5.7.14-log'  socket: ''  port: 50634  MySQL Community Server (GPL)
[INFO] port 50634 is available after 2s of waiting
[INFO] $ C:\Perforce_Workspace\ws3\InMemDb\target\mysql-dist\bin\mysqladmin.exe --no-defaults --wait=5 --port=50634 --user=root --socket=C:\Perforce_Workspace\ws3\InMemDb\target\mysql-data\mysql.sock --host=127.0.0.1 password root
[WARNING] >> Got error: Host 'localhost' is not allowed to connect to this MySQL server
[ERROR] #start('Config(tcpport=50634, dbuser=root, dbpassword=root, name=root, dbopts=[])', C:\Perforce_Workspace\ws3\InMemDb\target\mysql-dist, C:\Perforce_Workspace\ws3\InMemDb\target\mysql-data, true, NULL): thrown java.lang.IllegalArgumentException(Non-zero exit code 1: stdout = ; stderr = Got error: Host 'localhost' is not allowed to connect to this MySQL server
\n) out of com.jcabi.mysql.maven.plugin.MoreVerboseProcess#output[262] in 11s


3). When I run the same command through Command Prompt, with hostname as the name of my machine it works fine.
 C:\Perforce_Workspace\ws3\InMemDb\target\mysql-dist\bin\mysqladmin.exe --no-defaults --wait=5 --port=57085 --user=root --socket=C:\Perforce_Workspace\ws3\InMemDb\target\mysql-data\mysql.sock --host=[the hostname of my machine] password root

it says:
Waiting for MySQL server to answer...


The usage of 127.0.0.1 is hardcoded in the the jcabi-mysql-maven-plugin. Probably can we change it with usage of hostname -- or something that works. I have tried many other things like: bind-adrress setting, host file keeping 127.x.x.x entry --- but none of them worked. Request an early response.

In the https://github.com/jcabi/jcabi-mysql-maven-plugin/blob/cf29b4469634c5ac70365ef0ca043292242d8a0c/src/main/java/com/jcabi/mysql/maven/plugin/Instances.java the 127.x ip is hardcoded. Instead we can use: InetAddress.getLocalHost().getHostName()

**Comment by mike10004** [comment]
> 3.204.166.180	[the hostname of my machine]
> //127.0.0.1 localhost -- commented
> //::1 localhost -- commented

Removing localhost from your hosts file is an unusual configuration, and I can't replicate that setup, so I can't be much help with problems that result from that. If I recall correctly, MySQL's default security configuration privileges connections from localhost/127.0.0.1, so I would not be surprised that errors result from unconventional host reporting.

> In the https://github.com/jcabi/jcabi-mysql-maven-plugin/blob/cf29b4469634c5ac70365ef0ca043292242d8a0c/src/main/java/com/jcabi/mysql/maven/plugin/Instances.java the 127.x ip is hardcoded. Instead we can use: InetAddress.getLocalHost().getHostName()

Sure, give that a shot if you think it'll help. The process would be something like:
* fork the plugin from https://github.com/mike10004/jcabi-mysql-maven-plugin
* change the version in pom.xml
* make the code change you suggest
* build with `mvn install`
* change the dependency version in the pom of your project
* build and test your project
